### PR TITLE
Implemented prioritizing goals

### PR DESCRIPTION
**Pull request by JacekPelic**

### CODE REVIEW

File: client/masterAgent.py [hunk 1]
```diff
@@ -1,299 +1,307 @@
...
+        # Hence we if self.goalsInAction contains agent.goal this job has been finished so we need to remove it
+        for agent in agentsToReplan:
+            if agent.goal_details in self.goalsInAction:
+                self.goalsInAction.remove(agent.goal_details)
+
+        for goal in self.goalsInAction:
+            if goal in goalsToAssign:
+                goalsToAssign.remove(goal)
+
+        sortedGoals = self.priorytazeGoals(goalsToAssign)
```
Comment (Marouann) [inline client/masterAgent.py]: prioritizeGoals maybe

File: client/masterAgent.py [hunk 2]
```diff
@@ -1,299 +1,307 @@
...
+                # agt.current_plan.pop(0) = 'Action0'
+                # joint_action = ['NoOp', 'NoOp', 'NoOp', ...]
+                # joint_action[0] = ['agt0.Action0', 'NoOp', 'NoOp', ... ]
+                # joint_action[1] = ['agt0.Action0', 'agt1.Action0', 'NoOp', ... ] ...
+                joint_action[i] = agt.current_plan.pop(0)
+                # print("joint_action: ", file=sys.stderr, flush=True)
+            # print(joint_action, file=sys.stderr, flush=True)
+        return joint_action
+
+    def priorytazeGoals(self, goals):
```
Comment (Marouann) [inline client/masterAgent.py]: same prioritizeGoals

**Review by Marouann — CHANGES_REQUESTED** [review]

**Comment by Marouann** [comment]
Nice job, it's working very fine on SALazarus

File: client/state.py [hunk 3]
```diff
@@ -1,131 +1,177 @@
...
+            if atom.name == "GoalAt":
+                for location in neighbourLocations:
+                    if location == atom.variables[1]:
+                        neighbourGoals.append(atom)
+
+        return neighbourGoals
+
+    def getNeithbourFieldsWithoutGoals(self, position):
+        neighbourLocations = []
+        for atom in self.rigid_atoms:
```
Comment (Marouann) [inline client/state.py:115]: You are using two times the same 3 line of codes. You could create a method getNeighbourLocations into the class state

**Review by Marouann — COMMENTED** [review]
One thing that might be improved (later, not the priority right now) is to compute these prioritization at the initialization. Right now you are only using rigid_atoms so we could compute it once for all.

**Review by pierlj — APPROVED** [review]

**Pull request merged**